pipeline_0_annotate.sh

```diff
@@ -45,6 +26,60 @@ vep \
 --shift_3prime 1 \
 --no_escape \
 --numbers \
---fork 4
+--fork ${PARALLEL_THREADS}
 
 module unload VEP
+
+#CAPICE
+
+CAPICE_INPUT="${INPUT}"
+CAPICE_OUTPUT_DIR="${ANNOTATE_OUTPUT_DIR}"/step0b_capice
+
+if [[ ${OUTPUT_FILE} == *vcf ]]
+then
+    CAPICE_OUTPUT="${CAPICE_OUTPUT_DIR}"/"${OUTPUT_FILE/.vcf/.tsv}"
+    CAPICE_OUTPUT_VCF="${CAPICE_OUTPUT_DIR}"/"${OUTPUT_FILE/.vcf/.vcf.gz}"
+    CAPICE_OUTPUT_LOG="${CAPICE_OUTPUT}.log"
+else
+    CAPICE_OUTPUT="${CAPICE_OUTPUT_DIR}"/"${OUTPUT_FILE/.vcf.gz/.tsv}"
+    CAPICE_OUTPUT_VCF="${CAPICE_OUTPUT_DIR}"/"${OUTPUT_FILE}"
+    CAPICE_OUTPUT_LOG="${CAPICE_OUTPUT/.tsv/.log}"
+fi
+
+mkdir -p "${CAPICE_OUTPUT_DIR}"
+
+module load CADD
+# strip headers from input vcf for cadd
+gunzip -c $CAPICE_INPUT | sed '/^#/d' | bgzip > ${CAPICE_OUTPUT_DIR}/input_headerless.vcf.gz
+CADD.sh -a -g ${ASSEMBLY} -o ${CAPICE_OUTPUT_DIR}/cadd.tsv.gz ${CAPICE_OUTPUT_DIR}/input_headerless.vcf.gz
+module unload CADD
+
+module load CAPICE
+python ${EBROOTCAPICE}/CAPICE_scripts/model_inference.py \
+--input_path ${CAPICE_OUTPUT_DIR}/cadd.tsv.gz \
+--model_path ${EBROOTCAPICE}/CAPICE_model/xgb_booster.pickle.dat \
+--prediction_savepath ${CAPICE_OUTPUT} \
+--log_path ${CAPICE_OUTPUT_LOG}
+
+module load Java
+java -Djava.io.tmpdir="${TMPDIR}" -XX:ParallelGCThreads=2 -Xmx1g -jar capice2vcf.jar -i ${CAPICE_OUTPUT} -o ${CAPICE_OUTPUT_VCF}
```

Comment:
what is the reason to add ```-XX:ParallelGCThreads=2```? should this be included for the report jar as well?

Reply:
http://docs.gcc.rug.nl/gearshift/analysis/#q-why-are-my-java-jobs-misbehaving-and-using-more-cores-than-specified
Q: Why are my Java jobs misbehaving and using more cores than specified?
A: Tools written in Java sometimes use multiple cores. Usually they will allow you to specify how many threads/cores to use. To your surprise you may see your jobs use more cores than you specified. This happens because your Java application runs in a Java Virtual Machine (JVM) and the JVM itself also consumes resources. Recent versions of Java use an efficient garbage collection (GC) strategy that will detect how many cores are available on a computer and perform the cleanup using multiple threads in parallel. Hence if you specified your Java tool to use 4 cores on a node with 48 cores it is not uncommon to see spikes in CPU usage of 32+ cores when the garbage collector kicks in. When multiple of such Java jobs run on the same node this may result in overloading the node making it very slow and eventually crashing it. To limit the amount of threads Java should use for GC you can specify -XX:ParallelGCThreads=[integer]

Reply:
we should determine what is a sensible number...